packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -25,12 +33,32 @@ public static function yieldDirectory(string $directory, string $suffix = '*.php
      */
     private static function findFilesInDirectory(string $directory, string $suffix): array
     {
-        $finder = Finder::create()
-            ->in($directory)
-            ->files()
-            ->name($suffix);
+        $finder = Finder::create()->in($directory)->files()->name($suffix);
+        $fileInfos = iterator_to_array($finder);
+        $finderAll = Finder::create()->in($directory)->files();
+
+        foreach ($finderAll as $key => $fileInfoAll) {
+            $fileNameFromAll = $fileInfoAll->getFileName();
+            if (! isset($fileInfos[$key])) {
+                throw new ShouldNotHappenException(sprintf(
+                    '"%s" has invalid suffix, use "%s" suffix instead',
+                    $fileNameFromAll,
+                    $suffix
+                ));
+            }
+        }
 
+        return array_values($fileInfos);
+    }
+
+    /**
+     * @return SplFileInfo[]
+     */
+    private static function findFilesInDirectoryExclusively(string $directory, string $suffix): array
```

Comment:
This actually broke Rector tests: https://github.com/rectorphp/rector/runs/1801369216#step:11:102

Oh, just noticed the naming is inverted. Exlustive should be name for stricter version. 

I'll fix it so there is no BC Break

Reply:
Ref https://github.com/symplify/symplify/pull/2898

Reply:
Thank you 👍